### ISSUE TITLE
Dragging file between folders which match wildcards for different Build Action items results in file being included in both

### ISSUE THREAD

**Issue by dsplaisted** [dotnet/roslyn-project-system#1048]
- Create a new Console App (.NET Core) project
- Add the following to the `.csproj` file:

```xml
<ItemGroup>
  <None Include="**/*" Exclude="bin/**;obj/**;**/*.*proj;**/*.user" />
  <Content Include="wwwroot\**" />
  <None Remove="wwwroot\**" />
</ItemGroup>
```

- Create a `wwwroot` folder in the project folder
- Create `TextFile1.txt` in wwwroot folder
- Create `TextFile2.txt` in project root folder
- Alternatively, open the following repro project instead of the provious steps: [FolderBuildActionRepro.zip](https://github.com/dotnet/roslyn-project-system/files/675184/FolderBuildActionRepro.zip)
- In Solution Explorer, drag `TextFile2.txt` to the `wwwroot` folder
- Save project file and view it

EXPECTED: Move operation should have included the file in the `None` item and removed it from the `Content` item, ie:

```xml
  <ItemGroup>
    <Content Remove="wwwroot\TextFile2.txt" />
  </ItemGroup>
  <ItemGroup>
    <None Include="wwwroot\TextFile2.txt" />
  </ItemGroup>
```

ACTUAL: The file is included in the `None` item but not removed from `Content`.  So it will be in both items.  Further project modifications can result in more weirdness because of this.

```xml
  <ItemGroup>
    <None Include="wwwroot\TextFile2.txt" />
  </ItemGroup>
```

NOTE: Changing the Build Action of `TextFile1.txt` from `Content` to `None` does correctly remove the file from the `Content` item and include it in the `None` item.

**Comment by srivatsn** [comment]
Is this a CPS issue - @lifengl @jviau @adrianvmsft?

**Comment by adrianvmsft** [comment]
I think that behavior is by design... when you drag drop the file, the physical file gets removed from the source, so it doesn't need to be removed from Content anymore.

**Comment by jviau** [comment]
This is by design. You can see our globbing behavior document [here](https://github.com/Microsoft/VSProjectSystem/blob/master/doc/overview/globbing_behavior.md). When moving/renaming an item CPS' end goal is that the item remains included under its **current** item type. This means adding a `<None />` include for it. We do not care if there is an existing glob for another item type at the target location.

**Comment by dsplaisted** [comment]
@jviau 

> We do not care if there is an existing glob for another item type at the target location.

I don't think this matches user expectations.  If I drag an item from a folder covered by a `None` glob to a folder covered by a `Content` glob, I expect either:

- That the item will be included in `None` but not in `Content` (preserving the item type / build action)
- That the item will be included in `Content` but not in `None` (matching the default item type / build action applied to the destination)

I don't expect that the item will be included in both `None` and `Content` after the operation.
